node/node.go

```diff
@@ -396,9 +408,11 @@ func (n *Node) wsServerForPort(port int) *httpServer {
 }
 
 func (n *Node) stopRPC() {
-	n.http.stop()
-	n.ws.stop()
-	n.ipc.stop()
+	if !n.readOnly {
```

Comment:
Here and elsewhere, you're conflating "read-only" with "used-as-a-library" in a few places. You could imagine starting HTTP/WS/IPC services on a read-only geth process. Maybe `localLibrary` should be a separate option in Node?

And `readOnlyGeth/` should be renamed `readOnlyLib/`?